src/derivatives.h

```diff
@@ -67,51 +67,36 @@ std::map<std::string, matrix<Type>> derivatives(int n_visits, std::string cov_ty
 }
 // Base class of spatial and non-spatial derivatives.
 template <class Type>
-struct derivatives_base {
-  virtual matrix<Type> get_inverse(std::vector<int> visits, matrix<Type> dist) = 0;
+struct derivatives_base: virtual lower_chol_base<Type> {
   virtual matrix<Type> get_inverse_chol(std::vector<int> visits, matrix<Type> dist) = 0;
   virtual matrix<Type> get_sigma_derivative1(std::vector<int> visits, matrix<Type> dist) = 0;
   virtual matrix<Type> get_sigma_derivative2(std::vector<int> visits, matrix<Type> dist) = 0;
-  virtual matrix<Type> get_sigma(std::vector<int> visits, matrix<Type> dist) = 0;
   virtual matrix<Type> get_inverse_derivative(std::vector<int> visits, matrix<Type> dist) = 0;
+  virtual ~derivatives_base() {};
```

Comment:
needed because there are multiple inheritance, to avoid the destructor of base class being called

Reply:
Can you please add that as code comment?